samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -123,17 +154,60 @@ public synchronized String getEphemeralPath() {
   }
 
   /**
-   * Method is used to get the <i>sorted</i> list of currently active/registered processors
+   * Method is used to get the <i>sorted</i> list of currently active/registered processors (znodes)
    *
    * @return List of absolute ZK node paths
    */
   public List<String> getSortedActiveProcessors() {
-    List<String> children = zkClient.getChildren(keyBuilder.getProcessorsPath());
-    if (children.size() > 0) {
-      Collections.sort(children);
-      LOG.info("Found these children - " + children);
+    List<String> znodeIds = zkClient.getChildren(keyBuilder.getProcessorsPath());
+    if (znodeIds.size() > 0) {
+      Collections.sort(znodeIds);
+      LOG.info("Found these children - " + znodeIds);
+    }
+    return znodeIds;
+  }
+
+  /**
+   * Method is used to read processors PIDs
```

Comment:
this still reads "PIDs" . Please change it to "processor's data in Znode".  

Method name reads "getProcessorsPID". It is plural. But it returns only one processorId. The method is returning data for a znode path. It will be more clear to call the method name getProcessorData.